cernopendata_client/downloader.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+# This file is part of cernopendata-client.
+#
+# Copyright (C) 2020 CERN.
+#
+# cernopendata-client is free software; you can redistribute it and/or modify
+# it under the terms of the GPLv3 license; see LICENSE file for more details.
+
+from sys import stderr as STREAM
+import re
+import pycurl
+
+
+def show_download_progress(download_t, download_d, upload_t, upload_d):
+    """Show download progress of a file"""
+    kb = 1024
+    STREAM.write(
+        "Downloading: {}/{} kiB ({}%)\r".format(
+            str(int(download_d / kb)),
+            str(int(download_t / kb)),
+            str(int(download_d / download_t * 100) if download_t > 0 else 0),
+        )
+    )
+    STREAM.flush()
+
+
+def download_single_file(path=None, file_location=None):
+    """Download single file"""
+    file_name = file_location.split("/")[-1]
+    file_dest = path + "/" + file_name
+    with open(file_dest, "wb") as f:
+        print(
+            "==> Downloading file: ./{}/{}".format(
+                path,
+                file_name,
+            )
+        )
+        c = pycurl.Curl()
+        c.setopt(c.URL, file_location)
+        c.setopt(c.WRITEDATA, f)
+        c.setopt(c.NOPROGRESS, False)
```

Comment:
Observation: the progress bar, once a file is downloaded, makes the display broken a bit, see:

```console
$ cernopendata-client download-files --recid 1 --filter-range 1-2
==> Downloading file 1 of 2
==> Downloading file: ./1/0248915F-EE71-E011-8894-0025902009E8.root
==> Downloading file 2 of 2kiB (100%)
==> Downloading file: ./1/0268F635-B671-E011-9090-002481E14E00.root
Downloading: 842237/842237 kiB (100%)
Download completed!
```

Note the `Downloading file 2 of 2kiB (100%)` part where some bits remained from the previous progress bar.

Mentioning just for completeness; if you don't have a quick solution, we can also address this cosmetics issues later via another separate issue... 

However we can solve

Reply:
I tried to reproduce it with other `recid`s but it happens very rarely. I will find a workaround to stop this.